include/curl/websockets.h

```diff
@@ -30,13 +30,20 @@ extern "C" {
 
 struct curl_ws_frame {
   int age;              /* zero */
+  int actual_flags;      /* See the CURLWS_FLAGS_* defines */
   int flags;            /* See the CURLWS_* defines */
   curl_off_t offset;    /* the offset of this data into the frame */
   curl_off_t bytesleft; /* number of pending bytes left of the payload */
   size_t len;           /* size of the current data chunk */
 };
 
 /* flag bits */
+#define CURLWS_FLAGS_FIN   (1<<7)
+#define CURLWS_FLAGS_RSV1  (1<<6)
+#define CURLWS_FLAGS_RSV2  (1<<5)
+#define CURLWS_FLAGS_RSV3  (1<<4)
```

Comment:
I've just gone with the values that ultimately appear in the header hence the bit shifting starting at 4.

Reply:
One challenge: there are three RSV bits in every incoming WebSocket "frame", but the messages you get from libcurl may be a partial frame, a fragment, so we need to make sure that we return these bits set for all fragments of that frame in which the RSV bits are set.